### PR TITLE
Make computation parameters extendable

### PR DESCRIPTION
**Pull request by yichen88**
Signed-off-by: yichen88 <tang.yichenyves@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**
Now we can add specifical computation parameters as plugins

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
> Does this PR breaks something? Extending a new interface will break private implementations...
> If it breaks add the « breaking change » label to this PR



> Does this PR breaks something? Extending a new interface will break private implementations...
> If it breaks add the « breaking change » label to this PR

ComputationParameters has been added since last release,  so it depends what do we call "breaking change". In that case it does not break compatibility with last release.

File: computation/src/main/java/com/powsybl/computation/ComputationParameters.java [hunk 1]
```diff
@@ -30,24 +33,10 @@
      */
     OptionalLong getDeadline(String commandId);
 
-    Optional<String> getQos(String commandId);
-
+    /**
+     * @return an empty {@link ComputationParametersImpl}, but it supports plugin. See more {@link Extension}
```
Comment (sylvlecl) [inline computation/src/main/java/com/powsybl/computation/ComputationParameters.java]: A detail: the fact that it returns a "ComputationParametersIMPL" is an implementation detail, I think it's better to only mention the interface here.

File: computation/src/main/java/com/powsybl/computation/ComputationParametersImpl.java [hunk 2]
```diff
@@ -6,23 +6,25 @@
...
+public class ComputationParametersImpl extends AbstractExtendable<ComputationParameters> implements ComputationParameters {
 
     private final Map<String, Long> timeoutMap;
 
     private final Map<String, Long> deadlineMap;
 
-    private final Map<String, String> qosMap;
-
-    ComputationParametersImpl(Map<String, Long> timeoutMap, Map<String, Long> deadlineMap, Map<String, String> qosMap) {
+    ComputationParametersImpl(Map<String, Long> timeoutMap, Map<String, Long> deadlineMap) {
```
Comment (sylvlecl) [inline computation/src/main/java/com/powsybl/computation/ComputationParametersImpl.java]: A not important detail again, but maybe we should discuss this at powsybl level: should we have guidelines for naming ? @mathbagu ? 

In my opinion, it's not worth it to mention the type in the name of variables (timeoutMAP), same thing for the fields.
Maybe timeouts is enough (or timeoutsForCommand to explain what's the key part of the map).

File: computation/src/test/java/com/powsybl/computation/ComputationParametersTest.java [hunk 3]
```diff
@@ -51,4 +49,35 @@ public void testInvalid() {
...
+    public void testExt() {
+        // prepare
+        ComputationParameters base = ComputationParameters.empty();
+        QuantumComputationParameters quantum = new QuantumComputationParameters(42);
+        base.addExtension(QuantumComputationParameters.class, quantum);
+
+        // in quantum computation manager
+        QuantumComputationParameters quantumComputationParameters = base.getExtension(QuantumComputationParameters.class);
+        assertSame(quantum, quantumComputationParameters);
+        assertTrue(42 == quantumComputationParameters.getAtLeastQubits());
```
Comment (sylvlecl) [inline computation/src/test/java/com/powsybl/computation/ComputationParametersTest.java]: We should use assertEquals instead :)

Comment (yichen88) [reply computation/src/test/java/com/powsybl/computation/ComputationParametersTest.java]: > We should use assertEquals instead :)
Returned type is Integer not int.

File: computation/src/main/java/com/powsybl/computation/ComputationParametersImpl.java (hunk 2 above)
Comment (mathbagu) [reply computation/src/main/java/com/powsybl/computation/ComputationParametersImpl.java]: Agree: it’s a bad idea to suffix the name of the variable by its type. If the type change you will forget to fix the name of the variable.

File: computation/src/main/java/com/powsybl/computation/ComputationParameters.java [hunk 4]
```diff
@@ -30,24 +33,10 @@
      */
     OptionalLong getDeadline(String commandId);
 
-    Optional<String> getQos(String commandId);
```
Comment (mathbagu) [inline computation/src/main/java/com/powsybl/computation/ComputationParameters.java]: Does it break something?

Comment (yichen88) [reply computation/src/main/java/com/powsybl/computation/ComputationParameters.java]: > Does it break something?

Currently, no one else uses it, except me.

**Pull request merged**